File: jambeth/jAmbeth.Merge/src/main/java/de/osthus/ambeth/orm/LinkConfig.java
package de.osthus.ambeth.orm;

import de.osthus.ambeth.util.ParamChecker;

public class LinkConfig implements ILinkConfig
{
	protected String source;

	protected String alias;

	protected CascadeDeleteDirection cascadeDeleteDirection = CascadeDeleteDirection.NONE;

	public LinkConfig(String source)
	{
		ParamChecker.assertParamNotNull(source, "source");
		this.source = source;
	}

	protected LinkConfig()
	{
	}

	@Override
	public String getSource()
	{
		return source;
	}

	@Override
	public String getAlias()
	{
		return alias;
	}

	public void setAlias(String alias)
	{
		this.alias = alias;
	}

	@Override
	public CascadeDeleteDirection getCascadeDeleteDirection()
	{
		return cascadeDeleteDirection;
	}

	public void setCascadeDeleteDirection(CascadeDeleteDirection cascadeDeleteDirection)
	{
		this.cascadeDeleteDirection = cascadeDeleteDirection;
	}
}

File: ambeth/Ambeth.Merge/ambeth/orm/LinkConfig.cs
using System;
using De.Osthus.Ambeth.Util;

namespace De.Osthus.Ambeth.Orm
{
    public class LinkConfig : ILinkConfig
    {
        public String Source { get; private set; }

        public String Alias { get; set; }

        public CascadeDeleteDirection CascadeDeleteDirection { get; set; }

        public LinkConfig(String source)
            : this()
        {
            ParamChecker.AssertParamNotNull(source, "Source");
            this.Source = source;
        }

        protected LinkConfig()
        {
            CascadeDeleteDirection = CascadeDeleteDirection.NONE;
        }
    }
}
